cdhweb/pages/management/commands/exodus.py

```diff
@@ -219,3 +231,124 @@ def form_pages(self):
+                continue
+
+            # if image is in a subdirectory under uploads (e.g. projects, blog)
+            # add it to a collection with that name
+            relative_path = os.path.dirname(imgpath) \
+                .replace('%s/uploads/' % settings.MEDIA_ROOT, '')
+
+            # there are two variants of Slavic DH, one with and one
+            # without a space; remove the space so they'll be in one collection
+            basedir = relative_path.split('/')[0].replace(' ', '')
+            collection = None
+            if basedir:
+                collection = self.get_collection(basedir)
+
+            with open(imgpath, 'rb') as imgfilehandle:
+                title = os.path.basename(imgpath)
+                # passing collection=None errors, so
+                # only specify collection option when we have one
+                extra_opts = {}
+                if collection:
+                    extra_opts['collection'] = collection
+                try:
+                    Image.objects.create(
+                        title=title,
+                        file=ImageFile(imgfilehandle, name=title),
+                        **extra_opts)
+                except Exception as err:
+                    # seems to mean that height/width calculation failed
+                    # (usually non-images)
+                    print('Error creating image for %s: %s' % (imgpath, err))
+
+    def get_media_files(self):
+        # wagtail images support: GIF, JPEG, PNG, WEBP
+        imgfile_path = '%s/**/*.*' % settings.MEDIA_ROOT
+        # get filenames for all uploaded files
+        filenames = glob.glob(imgfile_path, recursive=True)
+        # aggregate files by basename to identify files with the same
+        # name in different locations
+        unique_filenames = defaultdict(list)
+        for path in filenames:
+            unique_filenames[os.path.basename(path)].append(path)
+
+        # check files with the same name in multiple locations
+        for key, val in unique_filenames.items():
+            if len(val) > 1:
+                samefile = filecmp.cmp(val[0], val[1], shallow=False)
+                # if the files are the same
+                if samefile:
+                    # keep the first one and remove the others from the
+                    # list of files to be migrated
+                    extra_copies = val[1:]
+
+                # if not all the same, identify the largest
+                # (all are variant/cropped versions of the same image)
+                else:
+                    largest_file = None
+                    largest_size = 0
+                    for filepath in val:
+                        size = os.stat(filepath).st_size
+                        if size > largest_size:
+                            largest_size = size
+                            largest_file = filepath
+
+                    extra_copies = [f for f in val if f != largest_file]
```

Comment:
did we ever end up with situations where there are totally unrelated images with the same name (e.g. `Untitled-1` or `Upload` or something generic)?

Reply:
ah, nvm - I see your report on the issue that confirms this isn't a problem.

Reply:
right — here's the list https://github.com/Princeton-CDH/cdh-web/issues/248#issuecomment-748287183